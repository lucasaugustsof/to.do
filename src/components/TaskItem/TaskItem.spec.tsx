import { describe, it, expect, vi } from 'vitest'

import { render, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'

import { TaskItem } from '.'

const onCompleteTask = vi.fn()
const onDeleteTask = vi.fn()

const testID = 'test:task-item'

describe('Components/TaskItem', () => {
  it('Should render the TaskItem component correctly', () => {
    const { asFragment } = render(
      <TaskItem
        title="Refactor my code"
        onCompleteTask={onCompleteTask}
        onDeleteTask={onDeleteTask}
      />,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should call the onCompleteTask function once', async () => {
    render(
      <TaskItem
        testID={testID}
        title="Refactor my code"
        onCompleteTask={onCompleteTask}
        onDeleteTask={onDeleteTask}
      />,
    )

    const checkboxComponent = screen.getByTestId(testID).children[0].children[0]
    await userEvent.click(checkboxComponent)

    expect(onCompleteTask).toBeCalledTimes(1)
  })

  it('should call the onDeleteTask function once', async () => {
    render(
      <TaskItem
        testID={testID}
        title="Refactor my code"
        onCompleteTask={onCompleteTask}
        onDeleteTask={onDeleteTask}
      />,
    )

    const deleteButtonComponent = screen.getByTestId(testID).children[1]
    await userEvent.click(deleteButtonComponent)

    expect(onCompleteTask).toBeCalledTimes(1)
  })
})
