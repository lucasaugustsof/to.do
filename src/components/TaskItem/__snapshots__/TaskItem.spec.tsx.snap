// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components/TaskItem > Should render the TaskItem component correctly 1`] = `
<DocumentFragment>
  <li
    class="_container_22b113"
  >
    <div>
      <button
        class="_container_9097fe"
      >
        <div
          data-checked="false"
        />
      </button>
      <p
        class="_taskTitle_22b113"
        data-completed="false"
      >
        Refactor my code
      </p>
    </div>
    <button
      class="_deleteButton_22b113"
    >
      <svg
        fill="currentColor"
        height="14"
        viewBox="0 0 256 256"
        width="14"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M216,48H176V40a24,24,0,0,0-24-24H104A24,24,0,0,0,80,40v8H40a8,8,0,0,0,0,16h8V208a16,16,0,0,0,16,16H192a16,16,0,0,0,16-16V64h8a8,8,0,0,0,0-16ZM96,40a8,8,0,0,1,8-8h48a8,8,0,0,1,8,8v8H96Zm96,168H64V64H192ZM112,104v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Zm48,0v64a8,8,0,0,1-16,0V104a8,8,0,0,1,16,0Z"
        />
      </svg>
    </button>
  </li>
</DocumentFragment>
`;
